src/inter_dc_repl_vnode.erl

```diff
@@ -64,42 +64,47 @@ init([Partition]) ->
 
 handle_command(trigger, _Sender, State=#state{partition=Partition,
                                               reader=Reader}) ->
-    {NewReaderState, Transactions} =
-        clocksi_transaction_reader:get_next_transactions(Reader),
     {ok, DCs} = inter_dc_manager:get_dcs(),
-    case Transactions of
-        [] ->
-            %% Send heartbeat
-            Heartbeat = [#operation
-                         {payload =
-                              #log_record{op_type=noop, op_payload = Partition}
-                         }],
-            DcId = dc_utilities:get_my_dc_id(),
-            {ok, Clock} = vectorclock:get_clock(Partition),
-            Time = clocksi_transaction_reader:get_prev_stable_time(NewReaderState),
-            TxId = 0,
-            %% Receiving DC treats hearbeat like a transaction
-            %% So wrap heartbeat in a transaction structure
-            Transaction = {TxId, {DcId, Time}, Clock, Heartbeat},
-            case inter_dc_communication_sender:propagate_sync(
-                   {replicate, [Transaction]}, DCs) of
-                ok ->
-                    NewReader = NewReaderState;
-                _ ->
-                    NewReader = NewReaderState
-            end;
-        [_H|_T] ->
-            case inter_dc_communication_sender:propagate_sync(
-                   {replicate, Transactions}, DCs) of
-                ok ->
-                    NewReader = NewReaderState;
-                _ ->
-                    NewReader = Reader
-            end
+    NewState = case DCs of
+        [] -> State;
+        DCs ->
+            {NewReaderState, Transactions} =
+                clocksi_transaction_reader:get_next_transactions(Reader),
+            case Transactions of
+                [] ->
+                    %% Send heartbeat
+                    Heartbeat = [#operation
+                                 {payload =
+                                      #log_record{op_type=noop, op_payload = Partition}
+                                 }],
+                    DcId = dc_utilities:get_my_dc_id(),
+                    {ok, Clock} = vectorclock:get_clock(Partition),
+                    Time = clocksi_transaction_reader:get_prev_stable_time(NewReaderState),
+                    TxId = 0,
+                    %% Receiving DC treats hearbeat like a transaction
+                    %% So wrap heartbeat in a transaction structure
+                    Transaction = {TxId, {DcId, Time}, Clock, Heartbeat},
+                    case inter_dc_communication_sender:propagate_sync(
+                           {replicate, [Transaction]}, DCs) of
+                        ok ->
+                            NewReader = NewReaderState;
+                        _ ->
+                            NewReader = NewReaderState
+                    end;
+                [_H|_T] ->
+                    case inter_dc_communication_sender:propagate_sync(
```

Comment:
It's better to do `Variable = case...` so you avoid warnings about exporting variables out of case statements.